### ISSUE TITLE
Failing System test on Mysqli CI 

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#8672]
* Reproduce: https://travis-ci.org/piwik/piwik/jobs/77998360
 * Fails only in the MYSQLI AllTests CI build.
 
```
1) Warning

The data provider specified for Piwik\Tests\System\AutoSuggestAPITest::testApi is invalid.

Unknown database 'piwik_tests'
```

maybe someone has some idea?

**Comment by diosmosis** [comment]
The AutoSuggestAPITest requires the plugins to be loaded for the data provider, which requires the database to exist. Data providers are run before setUp/setUpBeforeClass. For some reason on MYSQLi, the database doesn't exist here, but on PDO it does... Not sure why, but the best fix would be to remove the need to have the database for the data provider.

**Comment by mattab** [comment]
fixed in #8703
